sashimi-webapp/src/database/data-modifier/dataAdd.js

```diff
@@ -91,6 +93,28 @@ export default class dataAdd {
     );
   }
 
+  static duplicateFile(fileId) {
+    return new Promise((resolve, reject) => {
+      sqlCommands.getMaxFileId()
+      .then((maxId) => {
+        const newFileId = maxId + 1;
+        return sqlCommands.retrieveFullFile(fileId)
+        .then((file) => {
+          file[0].file_id = newFileId;
+          return file;
+        })
+        .then(duplicatedFile =>
+          sqlCommands.insertContent(constants.ENTITIES_FILE_MANAGER, duplicatedFile)
+          .then(() =>
+            resolve(duplicatedFile)
+          )
+          .catch(sqlError => reject(sqlError)))
+        .catch(sqlError => reject(sqlError));
+      })
+      .catch(sqlError => reject(sqlError));
+    });
+  }
+
```

Comment:
The reason `.all` can't be used is because the database only allows synchronous running otherwise the thread will be `blocked`. :face_with_head_bandage: